src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -359,6 +359,10 @@ int32_t GlobalizationNative_GetSortKeyNative(const uint16_t* localeName, int32_t
             return (int32_t)transformedStringBytes;
         NSRange range = NSMakeRange(0, [transformedString length]);
         NSUInteger usedLength = 0;
+        // If the buffer is too small, return the required buffer size
+        // and throw exception in managed code
+        if (cbSortKeyLength < (int32_t)transformedStringBytes)
```

Comment:
Nit. we could move this a few lines up to prevent the `NSRange` allocation.

Reply:
Thanks, updated.